vpr/src/pack/cluster.cpp

```diff
@@ -695,6 +742,32 @@ static void print_pack_status(int num_clb,
         VTR_LOG("\n");
         fflush(stdout);
         mols_since_last_print = 0;
+        if (attraction_groups.num_attraction_groups() > 0) {
+            update_attraction_group_status(attraction_groups);
+        }
+    }
+}
+
+/*
+ * Periodically update the attraction groups to reflect which atoms in them
+ * are still available for new clusters (i.e. remove the atoms that have already
+ * been packed from the attraction group).
+ */
+static void update_attraction_group_status(AttractionInfo& attraction_groups) {
+    auto& atom_ctx = g_vpr_ctx.atom();
+
+    for (int igroup = 0; igroup < attraction_groups.num_attraction_groups(); igroup++) {
+        AttractGroupId group_id(igroup);
+        AttractionGroup& group = attraction_groups.get_attraction_group_info(group_id);
+        AttractionGroup new_att_group_info;
+
+        for (AtomBlockId atom : group.group_atoms) {
+            if (atom_ctx.lookup.atom_clb(atom) == ClusterBlockId::INVALID()) {
```

Comment:
anything other than invalid means that it belongs to a cluster, been packed already